### PR TITLE
Initial work on fixing child interpolation

### PR DESCRIPTION
**Pull request by jimbaker**

### CODE REVIEW

File: src/tagstr_site/htm.py [hunk 1]
```diff
@@ -277,17 +278,19 @@ def interpolate(self, tag: AstNode) -> HTML:
 
 
 def html(*args: str | Interpolation) -> HTML:
-    parser = ASTParser()
-    for arg in args:
+    parser = AstParser()
+    for arg in enumerate(args):
```
Comment (pauleveritt) [inline src/tagstr_site/htm.py:282]: Why do `enumerate` here when you don't use the index position?

File: src/tagstr_site/htm.py [hunk 2]
```diff
@@ -300,14 +303,39 @@ def html(*args: str | Interpolation) -> HTML:
         '</body></html>',
     )
 
-    this_parser = ASTParser()
-    for this_arg in these_args:
-        this_parser.feed(this_arg)
+    parser = AstParser()
+    for arg in args:
```
Comment (pauleveritt) [inline src/tagstr_site/htm.py:307]: I was doing `this_arg` because PyCharm complains when you have variable names shadowing an outer scope.

**Comment by jimbaker** [comment]
The tests under `__main__` will be extracted in a separate PR, so I will merge this in as-is for now to get this fix in.

**Comment by pauleveritt** [comment]
@jimbaker I think you wanted to start over from scratch (last Saturday.) Do you want to keep this PR open?

**Comment by jimbaker** [comment]
Correct, I will close this PR out accordingly.

**Pull request closed without merging**